lib/net.ts

```diff
@@ -9,107 +9,129 @@ import * as https from 'https';
+		this.makeRequest();
 	}
-	encodeQuery(data: PostData) {
+	makeRequest() {
+		const protocol = url.parse(this.uri).protocol as string;
+		const net = protocol.includes('https:') ? https : http;
+		return new Promise(resolve => {
+			 this.request = net.get(this.opts ? this.opts : this.uri, response => {
+				  response.setEncoding('utf-8');
+				  const stream = new Streams.ReadStream({nodeStream: response});
+				  resolve({
+						stream: stream,
+						headers: response.headers,
+						statusCode: response.statusCode
+				  });
+			 });
+		});
+  }
+	encodeQuery(data: AnyObject) {
 		let out = '';
 		for (const key in data) {
 			if (out) out += `&`;
 			out += `${key}=${encodeURIComponent('' + data[key])}`;
 		}
 		return out;
 	}
-	/**
-	 * Makes a http/https get request to the given link and returns the status, headers, and a stream.
-	 * The request data itself can be read with ReadStream#read().
-	 * @param opts request opts - headers, etc.
-	 * @param body POST body
-	 */
-	getFullResponse(opts: NetRequestOptions = {}, body?: string | PostData): Promise<{
-		statusCode: number | undefined,
-		statusMessage: string | undefined,
-		headers: http.IncomingHttpHeaders | undefined,
-		stream: Streams.ReadStream,
-	}> {
-		if (opts.body) {
+	write(body: string) {
+		if (!this.request) throw new Error(`Writing to a request that doesn't exist.`);
+		const opts = this.opts;
+		if (opts?.body) {
 			if (body) throw new Error(`must not pass both body and opts.body`);
 			body = opts.body;
 		}
 
 		if (body && typeof body !== 'string') {
-			if (!opts.headers) opts.headers = {};
-			if (!opts.headers['Content-Type']) {
-				opts.headers['Content-Type'] = 'application/x-www-form-urlencoded';
+			if (!opts?.headers) opts!.headers = {};
+			if (!opts?.headers['Content-Type']) {
+				opts!.headers['Content-Type'] = 'application/x-www-form-urlencoded';
 			}
 			body = this.encodeQuery(body);
 		}
 
-		if (opts.query) {
+		if (opts?.query) {
 			this.uri += (this.uri.includes('?') ? '&' : '?') + this.encodeQuery(opts.query);
 		}
 
 		const postBody = body;
 		if (postBody) {
-			if (!opts.headers) opts.headers = {};
-			if (!opts.headers['Content-Length']) {
-				opts.headers['Content-Length'] = Buffer.byteLength(postBody);
+			if (!opts!.headers) opts!.headers = {};
+			if (!opts!.headers['Content-Length']) {
+				opts!.headers['Content-Length'] = Buffer.byteLength(postBody);
 			}
 		}
-
-		return new Promise((resolve, reject) => {
-			if (Config.noNetRequests) return reject(new Error(`Net requests are disabled.`));
-
-			const protocol = url.parse(this.uri).protocol as string;
-			const net = protocol.includes('https:') ? https : http;
-			const req = net.request(opts ? opts : this.uri, response => {
-				const stream = new Streams.ReadStream({nodeStream: response});
-				resolve({
-					statusCode: response.statusCode,
-					statusMessage: response.statusMessage,
-					headers: response.headers,
-					stream: stream,
-				});
-			});
```

Comment:
You deleted this old good code. :(

Your new code can't handle options or POST requests.